### PR TITLE
Add web3 service initial configuration

### PR DESCRIPTION
**Pull request by Mclovinn**

### CODE REVIEW

File: .sample-env [hunk 1]
```diff
@@ -11,4 +11,9 @@ TYPEORM_PORT=                   // 5432
 TYPEORM_SYNCHRONIZE=            // false
 TYPEORM_DROP_SCHEMA=            // false
 TYPEORM_RUN_MIGRATION=          // true
-TYPEORM_LOG=                    // false
\ No newline at end of file
+TYPEORM_LOG=                    // false
+
+NETWORK_ID                      // 4
+WALLET_PRIVATE_KEY              // undefined
```
Comment (NiicooR) [inline .sample-env:17]: if this values are mandatory we should use * insted of // undefined

File: .sample-env [hunk 2]
```diff
@@ -11,4 +11,9 @@ TYPEORM_PORT=                   // 5432
 TYPEORM_SYNCHRONIZE=            // false
 TYPEORM_DROP_SCHEMA=            // false
 TYPEORM_RUN_MIGRATION=          // true
-TYPEORM_LOG=                    // false
\ No newline at end of file
+TYPEORM_LOG=                    // false
+
+NETWORK_ID                      // 4
```
Comment (NiicooR) [inline .sample-env:16]: Maybe a doc reference to know what is the meaning of 4

File: src/config/env.config.ts [hunk 3]
```diff
@@ -47,8 +53,16 @@ export default () => {
     migrationsRun: process.env.TYPEORM_RUN_MIGRATION ? process.env.TYPEORM_RUN_MIGRATION === 'true' : true,
   };
 
+  const web3Config: Web3Config = {
+    networkId: parseInt(process.env.NETWORK_ID) || 4,
+    walletPrivateKey: `${process.env.WALLET_PRIVATE_KEY}`,
```
Comment (NiicooR) [inline src/config/env.config.ts:58]: I don't believe `${}` is necessary

File: src/web3/web3.service.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import * as ContractInstance from '../contracts/Example.json';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const Web3 = require('web3');
```
Comment (NiicooR) [inline src/web3/web3.service.ts:6]: Could Web3 be imported with import sintax instead of require?

File: src/web3/web3.service.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  '137': 'https://polygon-mainnet.infura.io/v3/',
+};
+
+@Injectable()
+export class Web3Service {
+  private readonly web3;
+  private readonly walletAccount;
+  private readonly publicKey;
+  private readonly deployedNetwork;
+  private readonly nftFactoryInstance;
```
Comment (NiicooR) [inline src/web3/web3.service.ts:21]: is posible to add a type for this references?

File: src/web3/web3.service.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    this.nftFactoryInstance = new this.web3.eth.Contract(
+      ContractInstance.abi,
+      this.deployedNetwork && this.deployedNetwork.address,
+    );
+  }
+
+  async getTokenInformation(tokenUrl: string): Promise<any> {
+    const tokenInfo = await this.nftFactoryInstance.methods.approvedUris(tokenUrl).call();
+
+    return tokenInfo;
```
Comment (NiicooR) [inline src/web3/web3.service.ts:42]: If tokenInfo is typed we are overriding the inferred return type with Promise<any> in the method signature. We should delete Promise<any> and let typescript to inferred the type of the method if It's possible.

File: src/web3/web3.service.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  decodeEvent(data) {
+    const hexString = data.logs[2].data;
```
Comment (NiicooR) [inline src/web3/web3.service.ts:46]: 2 is a magic number, could we add a comment that explains why 2 and if define a constant for this number.

**Review by NiicooR — APPROVED** [review]

File: src/web3/web3.service.ts (hunk 4 above)
Comment (Mclovinn) [reply src/web3/web3.service.ts:6]: I had a lot of problems trying to import the lib using import syntax. I really don't know why I couldn't make it work

File: src/web3/web3.service.ts (hunk 5 above)
Comment (Mclovinn) [reply src/web3/web3.service.ts]: I'll try haha

**Pull request merged**